docs/net.md

```diff
@@ -0,0 +1,269 @@
+`void onclose(void)`
+
+`close` event emitted when the socket has been closed, either by you or by
+the remote end.
+
+### Socket.onconnect Event
+`void onconnect(void)`
+
+`connect` event emitted when the socket has made a successful connection to a
+remote TCP server.
+
+### Socket.ondata Event
+`void ondata(Buffer buf)`
+
+`data` event emitted when the socket has received data. `buf` is a Buffer
+containing the data received.
+
+### Socket.onerror Event
+`void onerror(void)`
+
+`error` event emitted when there was an error on the socket (read/write/connect).
+
+### Socket.ontimeout Event
+`void ontimeout(void)`
+
+`timeout` event emitted when the socket has timed out. This can only happen
+if a timeout was set with `setTimeout`.
+
+### Socket.connect
+`void connect(Object options, callback onconnect)`
+
+Connect to a remote TCP server.
+
+`options` is an object which specifies:
+```
+{
+    port : Port the client should connect to (required)
+    host : Host the client should connect to
+    localAddress : Local address to bind to
+    localPort : local port to bind to
+    family : Version of IP stack, deafults to 4
+}
+```
+
+`onconnect` is optional and, if specified, will be added as the listener for the
+`connect` event.
+
+### Socket.pause
+`void pause(void)`
+
+Pause a socket from receiving data. `data` event will not be emitted until
+`resume` is called.
+
+### Socket.resume
+`void resume(void)`
+
+Allow a socket to resume receiving data after a call to `pause`.
+
+### Socket.setTimeout
+`void setTimeout(Number time, callback ontimeout)`
+
+Set a socket timeout. This will start a timer on the socket which will expire
+in `time` milliseconds if there has been no activity on the socket. The
+`ontimeout` parameter, if specified, will register a listener for the
+`timeout` event.
+
+### Socket.write
+`void write(Buffer buf, callback writeDone)`
+
+Send data on the socket.
+
+`buf` should contain the data you wish to send.
+
+`writeDone` is optional and will be called once the data is written.
+
+### Server.onclose Event
+`void onclose(void)`
+
+`close` event emitted when the server has closed. This only happens after a
+server has called `close()` and all its connections have been closed. Calling
+`close()` does not close all opened connections; that must be done manually.
+
+### Server.onconnection Event
+`void onconnection(Socket sock)`
+
+`connection` event emitted when a client has connected to the server.
+
+`sock` parameter is the socket for this new connection.
+
+### Server.onerror Event
+`void onerror(void)`
+
+`error` event emitted when the server has had an error.
+
+### Server.onlistening Event
+`void onlistening(void)`
+
+`listening` event emitted when the server has been bound, after calling
+`server.listen()`.
+
+### Server.address
+`Object address(void)`
+
+Returns an address object for the server:
+```
+{
+    port : Server port
+    family : IPv4 or IPv6
+    address : IP address for the server
+}
```

Comment:
This options object and the one below should have types specified for the fields. Maybe makes sense to define this as AddressOptions and move def to the webidl section above?